16_wordle/main.go

```diff
@@ -16,40 +20,43 @@ func main() {
 		log.Fatal(err)
 	}
 
-	validWords := bytes.Split([]byte(rawWordList), []byte("\n"))
-	chosenWord := randomWord(validWords)
+	wordList := bytes.Split([]byte(rawWordList), []byte("\n"))
+
+	validWords := List{
+		words: wordList,
+	}
```

Comment:
how about adding another function to the `List` type:

```go
func NewList(path string) (List, error){
  rawWordList, err := os.ReadFile("word_list/word_list")
	// ...

	wordList := bytes.Split([]byte(rawWordList), []byte("\n"))

  // ...
}

```

https://go.dev/blog/error-handling-and-go

Errors are a very important part of the Go language, and understanding how they work is important :)

Reply:
BTW: feel free to add it in another PR